src/objective-c/GRPCClient/private/NSDictionary+GRPC.m

```diff
@@ -53,4 +54,24 @@ + (instancetype)grpc_dictionaryFromMetadata:(grpc_metadata *)entries count:(size
   }
   return metadata;
 }
+
+- (size_t)grpc_toMetadataArray:(grpc_metadata **)metadata {
+  size_t count = 0;
+  size_t capacity = 0;
+  for (id key in self) {
+    capacity += [self[key] count];
+  }
+  *metadata = gpr_malloc(capacity * sizeof(grpc_metadata));
+  for (id key in self) {
+    id value_array = self[key];
```

Comment:
Please check GRPCCall.h's documentation for the structure of the metadata dictionary. It's not an map NSString -> NSString[], but a map NSString -> (NSString | NSData). You can see how to properly handle it in the previous implementation of GRPCCall::sendHeaders (line 248).